src/components/views/create/WalletAddressInput.tsx

```diff
@@ -31,18 +44,66 @@ const WalletAddressInput = (props: {
 	} = useFormContext();
 
 	const [isHidden, setIsHidden] = useState(false);
-
-	const { networkId, defaultValue, sameAddress, isActive, setIsActive } =
-		props;
+	const [isValidating, setIsValidating] = useState(false);
 
 	const { chainId, library } = useWeb3React();
 
 	const user = useAppSelector(state => state.user?.userData);
 	const isGnosis = networkId === config.SECONDARY_NETWORK.id;
-	const isDefaultAddress = compareAddresses(
-		getValues(isGnosis ? EInputs.secondaryAddress : EInputs.mainAddress),
-		user?.walletAddress,
+	const value = getValues(
+		isGnosis ? EInputs.secondaryAddress : EInputs.mainAddress,
 	);
+	const isDefaultAddress = compareAddresses(value, user?.walletAddress);
+
+	let disabled: boolean;
+	if (isGnosis) disabled = !isActive;
+	else disabled = !isActive && !sameAddress;
+
+	const isPrevAddress = (i: string) => {
+		if (userAddresses.length === 0) return false;
+		return userAddresses
+			.map(i => i.toLowerCase())
+			.includes(i.toLowerCase());
+	};
```

Comment:
please use meaningful names for variables.
has 2 variables with the same name in parent and child scope may confusing.